org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/model/JdiRequestFactory.scala

```diff
@@ -54,23 +55,24 @@ object JdiRequestFactory {
   /**
    * create a line breakpoint at the given line, if available
    */
-  def createBreakpointRequest(referenceType: ReferenceType, lineNumber: Int, debugTarget: ScalaDebugTarget, suspendPolicy: Int): Option[BreakpointRequest] = {
-    val locations = JDIUtil.referenceTypeToLocations(referenceType)
-    // TODO: is it possible to have the same line number in multiple locations? need test case
-    // see #1001370
-    val line = locations.find(_.lineNumber == lineNumber)
-    line.map {
-      l =>
-        val breakpointRequest = debugTarget.virtualMachine.eventRequestManager.createBreakpointRequest(l)
-        breakpointRequest.setSuspendPolicy(suspendPolicy)
-        breakpointRequest
+  def createBreakpointRequest(referenceType: ReferenceType, lineNumber: Int, debugTarget: ScalaDebugTarget, suspendPolicy: Int): Option[BreakpointRequest] =
+    jdiSynchronized {
+      val locations = JDIUtil.referenceTypeToLocations(referenceType)
+      // TODO: is it possible to have the same line number in multiple locations? need test case
+      // see #1001370
+      val line = locations.find(_.lineNumber == lineNumber)
+      line.map {
+        l =>
+          val breakpointRequest = debugTarget.virtualMachine.eventRequestManager.createBreakpointRequest(l)
+          breakpointRequest.setSuspendPolicy(suspendPolicy)
+          breakpointRequest
+      }
     }
-  }
 
   /**
    * Create thread start request
    */
-  def createThreadStartRequest(virtualMachine: VirtualMachine): ThreadStartRequest = {
+  def createThreadStartRequest(virtualMachine: VirtualMachine): ThreadStartRequest = jdiSynchronized {
```

Comment:
Are you sure none of that methods is called inside actor?
Because in that case you would need to pack it into Future.
Just asking :)